### PR TITLE
Nuke resumable concept and auto continue from celery failure only if AUTO_START

### LINKED ISSUE #8307: Async migrations - nuke resumable concept
## Is your feature request related to a problem?

All migrations should be resumable as decided in https://github.com/PostHog/product-internal/blob/main/requests-for-comments/2022-01-04-async-migrations-improvements.md#upgrades-killing-async-migrations so lets remove that as an option to avoid failures in the future with forgetting that flag

## Describe alternatives you've considered

Keeping it as-is, maybe we'll need it in the future (if we do we can re-add it, the code is more complex atm for no good reason, plus we'd really need to think about the upgrades killing migrations problem before considering re-adding it).

### PR DESCRIPTION
**Pull request by tiina303**
## Changes

Closes https://github.com/PostHog/posthog/issues/8307

## How did you test this code?

CI

### CODE REVIEW

**Comment by yakkomajuri** [comment]
q: do we _need_ to nuke resumable? 

I do get the point - everything should be resumable.

But what if I want to have a function in there that should not resume (if Celery crashes it should just error)?

With the addition of `fn` this is something that could happen. Wondering what the benefit of getting rid of this completely is beyond just reducing a tiny bit of complexity?

**Comment by tiina303** [comment]
Reducing complexity is a very good reason to get rid of stuff. We agreed on the RFC that we'll make everything resumable, if we want to have anything non-resumable we'll need to solve upgrades killing the migration first. Having that flag as an option makes the migration writer think it's ok to have non-resumable migrations.

Currently if you want something that shouldn't resume you can do so with the function, e.g sth like (you shouldn't but you could):
```
if started_in_the_past():  // e.g. via constance config specifically created for this migration step
  raise Exception("this isn't resumable")
mark_as_started()
// whatever the fn is supposed to do
```

File: posthog/tasks/async_migrations.py [hunk 1]
```diff
@@ -59,7 +58,7 @@ def check_async_migration_health() -> None:
 
     # the worker crashed - this is how we find out and process the error
     if migration_instance.celery_task_id not in active_task_ids:
-        if is_current_operation_resumable(migration_instance):
+        if getattr(config, "AUTO_START_ASYNC_MIGRATIONS"):
```
Comment (yakkomajuri) [inline posthog/tasks/async_migrations.py:61]: are you sure we want this here? If you're removing the resumable concept then we should just try to resume automatically, no? I see `AUTO_START_ASYNC_MIGRATIONS` as a flag that's out of scope here.

Comment (tiina303) [reply posthog/tasks/async_migrations.py:61]: will do a separate conf

**Pull request merged**